### PR TITLE
 Update SECURITY.md and remove support for 2.x version

### PR DESCRIPTION
**Pull request by ernestognw**
With the new 5.x release, we can remove security fixes support for versions like 2.x
Also, adds support for 5.x release and it'll show up in the [Security Policy](https://github.com/OpenZeppelin/openzeppelin-contracts/security/policy) once merged.

#### PR Checklist

- [ ] Tests
- [ ] Documentation
- [ ] Changeset entry (run `npx changeset add`)

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -30,13 +30,14 @@ Only critical severity bug fixes will be backported to past major releases.
 
 | Version | Critical security fixes | Other security fixes |
 | ------- | ----------------------- | -------------------- |
-| 4.x     | :white_check_mark:      | :white_check_mark:   |
+| 5.x     | :white_check_mark:      | :white_check_mark:   |
+| 4.9     | :white_check_mark:      | :x:                  |
```
Comment (ernestognw) [inline SECURITY.md:34]: Do you agree with this? @Amxx
We probably would like to keep medium severity fixes in 4.9 along with the critical for some time.

File: SECURITY.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ Security vulnerabilities should be disclosed to the project maintainers through
 
 Responsible disclosure of security vulnerabilities is rewarded through a bug bounty program on [Immunefi].
 
-There is a bonus reward for issues introduced in release candidates that are reported before making it into a stable release.
+There is a bonus reward for issues introduced in release candidates that are reported before making it into a stable release. Learn more about release candidates at [Releases](../RELEASING.md).
```
Comment (ericglau) [inline SECURITY.md:11]: Suggestion for style consistency with other documents and to fix broken link
```suggestion
There is a bonus reward for issues introduced in release candidates that are reported before making it into a stable release. Learn more about release candidates at [`RELEASING.md`](./RELEASING.md).
```

**Review by ericglau — APPROVED** [review]

**Comment by ernestognw** [comment]
The community keeps referencing the security page and we should communicate correctly what are we supporting. I'm merging.

**Comment by Ogruj098** [comment]
> With the new 5.x release, we can remove security fixes support for versions like 2.x Also, adds support for 5.x release and it'll show up in the [Security Policy](https://github.com/OpenZeppelin/openzeppelin-contracts/security/policy) once merged.
> 
> #### PR Checklist
> * [x]  Tests
> * [x]  Documentation
> * [x]  Changeset entry (run `npx changeset add`)

**Pull request merged**